### PR TITLE
fix(render/ids): render ad api with privilege user if persistent storage is used

### PR DESCRIPTION
**Pull request by freecaykes**
Since we require  AD API Pods to write to certain static paths in the volume storage, because of the unprivileged PSS, we now need to use a privileged user

## For PR author

- [x] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/render/intrusion_detection.go
Comment (hjiawei) [inline pkg/render/intrusion_detection.go:74]: It is more common to use `/media` to mount storage type devices as documented in https://tldp.org/LDP/Linux-Filesystem-Hierarchy/html/media.html. How difficult is it to change this to `/media/ADAPIObjectName/storage`?

**Comment by hjiawei** [comment]
On question on the mount path but LGTM over all.

File: pkg/render/intrusion_detection.go
Comment (freecaykes) [reply pkg/render/intrusion_detection.go]: udpated to using `/mnt/...` https://github.com/tigera/operator/pull/2300/commits/ea759041f6c39bb7b8c7c7eec2a90a7e0cc69b7b

File: pkg/render/intrusion_detection.go
Comment (hjiawei) [inline pkg/render/intrusion_detection.go:1426]: Do you want to set `Provileged` to be openshift only like here. otherwise, leave it non-privileged?

https://github.com/tigera/operator/blob/c0be4fb5cc3535b3404a21b7d2895ea9cfef3b43/pkg/render/intrusion_detection.go#L642

Comment (freecaykes) [reply pkg/render/intrusion_detection.go]: just going to set the privileged to `false` https://github.com/tigera/operator/pull/2300/commits/19a50c7bc33cd00d8728c3eba55b27501675bb0e - tested in openshift doesnt need to be priivileged

**Review by hjiawei — APPROVED** [review]

File: pkg/render/intrusion_detection.go
Comment (rene-dekker) [inline pkg/render/intrusion_detection.go:1410]: Looks like this func no longer needs the bool passed in, since there is a new method.

File: pkg/render/intrusion_detection.go
Comment (rene-dekker) [inline pkg/render/intrusion_detection.go:1422]: Isn't it /mnt?

Comment (freecaykes) [reply pkg/render/intrusion_detection.go]: updated comments https://github.com/tigera/operator/pull/2300/commits/2932b4024aeb3e547363a5f0421635d82eb7ebae

File: pkg/render/intrusion_detection.go
Comment (freecaykes) [reply pkg/render/intrusion_detection.go]: switched to using the method call instead https://github.com/tigera/operator/pull/2300/commits/27a659534b6ddf2f6390379f8f3f515f4721574a

**Review by rene-dekker — APPROVED** [review]
lgtm

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Pull request merged**
